### ISSUE TITLE
failed to locate @import file in windows

### ISSUE THREAD

**Issue by GottZ** [stylus/stylus#356]
this is a dump of what my console says when i request /stylesheets/style.css within windows 7
express and jade work fine.

this is basicly the socket.io chat example wich i try running on windows.

```
d:\node\chat>node app.js
   app listening on http://0.0.0.0:12345
Error: D:\node\chat\public\stylesheets\style.styl:119
   115|     right 5px
   116|
   117| button
   118|   download-button()
 > 119|

failed to locate @import file d:\node\node_modules\stylus\lib/functions.styl

    at Evaluator.visitImport (d:\node\node_modules\stylus\lib\visitor\evaluator.js:567:21)
    at Evaluator.<anonymous> (d:\node\node_modules\stylus\lib\visitor\index.js:29:39)
    at Evaluator.visit (d:\node\node_modules\stylus\lib\visitor\evaluator.js:70:18)
    at Evaluator.<anonymous> (d:\node\node_modules\stylus\lib\visitor\evaluator.js:99:10)
    at Array.forEach (native)
    at Evaluator.setup (d:\node\node_modules\stylus\lib\visitor\evaluator.js:96:16)
    at Evaluator.evaluate (d:\node\node_modules\stylus\lib\visitor\evaluator.js:138:8)
    at Renderer.render (d:\node\node_modules\stylus\lib\renderer.js:50:26)
    at d:\node\node_modules\stylus\lib\middleware.js:148:17
    at [object Object].<anonymous> (fs.js:108:5)
```

i edited the chat example because require.paths does not work anymore it seems.

i simply commented this line out:
_require.paths.unshift(__dirname + '/../../lib/');_
though i think this has nothing to do with stylus.

well the problem above does.

i will debug / try what ever you want. just tell me what you want to know incase you dont have a windows test enviroment for this.

**Comment by matthew-dean** [comment]
I just ran into this issue. Similar to @extempl, Stylus fails when importing from a network path. As in:

```
@import 'lib/kouto-swiss/'
```
On Windows, this fails with:
```
Cannot find module '\\psf\Home\git\kouto-swiss-master\node_modules\lib\kouto-swiss\package.json'
```
Windows compiling succeeds if the folder is copied locally.

**Comment by matthew-dean** [comment]
Can you re-open?

**Comment by Panya** [comment]
Well, from the error message I can say that it's different problem. Stylus tried to find `package.json` file because didn't find `index.styl` inside this dir (`lib/kouto-swiss/index.styl`), but did not succeed. Can you show files structure for `lib/kouto-swiss/`(`tree lib/kouto-swiss`)?

**Comment by Panya** [comment]
Looks like the same bug as in https://github.com/isaacs/node-glob/issues/74 (we're using `node-glob` to find files for imports).

**Comment by matthew-dean** [comment]
I'm using this repo as-is: https://github.com/krkn/kouto-swiss

**Comment by sibbl** [comment]
I had the same problem on Ubuntu and the problem was that the file path had special characters like "[]!" in it. Moving the directory somewhere else solved it. Perhaps this will help someone.

**Comment by julmot** [comment]
Have this issue currently

**Comment by MartinKavik** [comment]
Problem with path `C:\abrigo\new_projects\FusionD-docs\.vuepress\override.styl` on Windows 10. 

_Workaround_: Change `new_projects` to `_new_projects`, because `\n` creates new line in the import path.

**Comment by meteorlxy** [comment]
@matthew-dean 

How to deal with those files in `node_modules` ?

**Comment by cedx** [comment]
I also have this issue. When compiling a dead simple stylesheet, on Linux everything is fine. But on Windows (my main workstation), I get the following error:

```
C:\Projets\mc2it\console\node_modules\stylus\bin\stylus:684
              throw err;
              ^

Error: src\mc2it_console\ui\main.styl:2:14
   1| body
   2|   color: white
-------------------^
   3|

failed to locate @import file C:\Projets\mc2it\console\node_modules\stylus\lib\functions\index.styl

    at Evaluator.visitImport (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\evaluator.js:915:21)
    at Visitor.visit (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\index.js:28:40)
    at Evaluator.visit (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\evaluator.js:160:18)
    at Evaluator.visitRoot (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\evaluator.js:707:27)
    at Visitor.visit (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\index.js:28:40)
    at Evaluator.visit (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\evaluator.js:160:18)
    at Evaluator.evaluate (C:\Projets\mc2it\console\node_modules\stylus\lib\visitor\evaluator.js:247:15)
    at Renderer.render (C:\Projets\mc2it\console\node_modules\stylus\lib\renderer.js:86:26)
    at C:\Projets\mc2it\console\node_modules\stylus\bin\stylus:678:15
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  lineno: 2,
  column: 14,
  filename: 'src\\mc2it_console\\ui\\main.styl',
  stylusStack: '',
  input: 'body\r\n\tcolor: white\r\n'
}
```

Node.js 18.1.0 on Windows 10 + Stylus 0.57.0

**Comment by cedx** [comment]
Ok, my problem is related to #2680 : downgrading `glob` to v7.2.0 fixes the issue.

**Comment by isaacs** [comment]
Should be fixed by glob@7.2.3.  If it isn't, please post a new issue at https://github.com/isaacs/node-glob/issues, ideally with a reproduction case.  Thanks!
